src/lib.rs

```diff
@@ -51,6 +51,9 @@ pub mod treemeta;
 /// This module contains TreeNode.
 pub mod treenode;
 
+/// Errors
+pub mod error;
```

Comment:
this appears to reference error.rs, which is not in the PR.  Does it compile?

Reply:
Ah missed this, will sort it now.

Reply:
Perhaps we need to define the sort criteria as you had it but reverse it then pop or 

```rust
v.sort()
v.reverse()
v.pop()
```